src/backend/renderer/element/memory.rs

```diff
@@ -0,0 +1,278 @@
+//! TODO: docs
+
+use std::{
+    any::TypeId,
+    collections::{hash_map::Entry, HashMap},
+    sync::{Arc, Mutex, MutexGuard},
+};
+
+use slog::trace;
+
+use crate::{
+    backend::renderer::{
+        utils::{CommitCounter, DamageTracker},
+        Frame, ImportMem, Renderer,
+    },
+    utils::{Buffer, Logical, Physical, Point, Rectangle, Size, Transform},
+};
+
+use super::{Id, RenderElement};
+
+#[derive(Debug, Default)]
+struct MemoryRenderBufferInner {
+    mem: Vec<u8>,
+    damage_tracker: DamageTracker<i32, Buffer>,
+    textures: HashMap<(TypeId, usize), Box<dyn std::any::Any>>,
+    renderer_seen: HashMap<(TypeId, usize), CommitCounter>,
+}
+
+impl MemoryRenderBufferInner {
+    fn from_memory(mem: &[u8]) -> Self {
```

Comment:
Maybe add a little todo here, that we want to allow specifying the color format in the future. (Right now the import/update_buffer calls don't support that anyway.)

Reply:
I refactored the `memory` again to already include the `Size` of the buffer instead of the raw `usize` and added a `TODO` for the format